library/common/main_interface.h

```diff
@@ -83,6 +83,14 @@ envoy_status_t reset_stream(envoy_stream_t stream);
  */
 envoy_engine_t init_engine();
 
+/**
+ * Update the network interface to the preferred network for opening new streams.
+ * Note that this state is shared by all engines.
```

Comment:
Is this permanent? Seems like ideally we'd be able to set this per-engine (especially since we don't expose a singleton on the platform layer).

Reply:
The presumption here is that the same networks/interfaces are available to any thread in the process. As long as that's true, there's no real inherent need for the state to be replicated.

I could see per-instance state being slightly more testable, maybe.